lib/node_modules/@stdlib/math/generics/statistics/incrprod/README.md

```diff
@@ -0,0 +1,123 @@
+# incrprod
+
+> Compute a product incrementally.
+
+
+<section class="usage">
+
+## Usage
+
+``` javascript
+var incrprod = require( '@stdlib/math/generics/statistics/incrprod' );
+```
+
+#### incrprod()
+
+Returns an accumulator `function` which incrementally computes a product.
+
+``` javascript
+var accumulator = incrprod();
+```
+
+#### accumulator( \[x\] )
+
+If provided an input value `x`, the accumulator function returns an updated product. If not provided an input value `x`, the accumulator function returns the current product.
+
+``` javascript
+var product = accumulator( 2.0 );
+// returns 2.0
+
+product = accumulator( 1.0 );
+// returns 3.0
+
+product = accumulator( 3.0 );
+// returns 6.0
+
+product = accumulator();
+// returns 6.0
+```
+
+</section>
+
+<!-- /.usage -->
+
+
+<section class="notes">
+
+## Notes
+
+* Input values are __not__ type checked. If provided `NaN` or a value which, when used in computations, results in `NaN`, the accumulated value is `NaN` for __all__ future invocations. If non-numeric inputs are possible, you are advised to type check and handle accordingly __before__ passing the value to the accumulator function.
+* For long running accumulations or accumulations of large numbers, care should be taken to prevent overflow. Overflow need not be forever since internally the accumulator stores the product as a __mantissa__ and __exponent__.
+
+
+</section>
+
+<!-- /.notes -->
+
+
+<section class="examples">
+
+## Examples
+
+### Default usage
+
+``` javascript
+var randu = require( '@stdlib/math/base/random/randu' );
+var incrprod = require( '@stdlib/math/generics/statistics/incrprod' );
+
+var accumulator;
+var v;
+var i;
+
+// Initialize an accumulator:
+accumulator = incrprod();
+
+// For each simulated value, update the product...
+for ( i = 0; i < 100; i++ ) {
+    v = randu() * 100.0;
+    accumulator( v );
+}
+console.log( accumulator() );
+```
+
+### Overflow can vanish
+
+``` javascript
+'use strict'
+
+var ldexp = require('@stdlib/math/base/special/ldexp');
+var incrprod = require('@stdlib/math/generics/statistics/incrprod');
+
+/* Write huge numbers x and y as a double starting from a normalized mantissa and big exponent */
+var x = ldexp( 0.5, 1000 );
+var y = ldexp( 0.1, 1000 );
+
+/* The number z on the other hand is practically zero */
+var z = ldexp( 0.2 , -1000 );
+
+var accumulator = incrprod();
+
+console.log( "x has a value\t" + x );
```

Comment:
Rather than logging to the console here,

``` javascript
var x = ldexp( 0.5, 1000 );
// returns <value>

var y = ldexp( 0.1, 1000 );
// returns <value>

// ...

var out = acc( x );
// returns <value>

out = acc( y );
// returns <value>

out = acc( z );
// returns <value>
```
where `<value>` indicates the actual numeric value.